crates/oxc_linter/src/rules/eslint/no_script_url.rs

```diff
@@ -0,0 +1,98 @@
+use oxc_ast::AstKind;
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("eslint(no-script-url): Script URL is a form of eval")]
+#[diagnostic(severity(warning), help("Disallow `javascript:` urls"))]
+struct NoScriptUrlDiagnostic(#[label] pub Span);
+
+#[derive(Debug, Default, Clone)]
+pub struct NoScriptUrl;
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// Disallow javascript: urls
+    ///
+    /// ### Why is this bad?
+    /// Using javascript: URLs is considered by some as a form of eval. Code passed in javascript: URLs has to be parsed and evaluated by the browser in the same way that eval is processed.
+    ///
+    /// ### Example
+    /// ```javascript
+    /// /*eslint no-script-url: "error"*/
+    ///
+    /// location.href = "javascript:void(0)";
+    ///
+    /// location.href = `javascript:void(0)`;
+    /// ```
+    NoScriptUrl,
+    style
+);
+
+impl Rule for NoScriptUrl {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        match node.kind() {
+            AstKind::StringLiteral(literal)
+                if literal.value.to_lowercase().starts_with("javascript:") =>
+            {
+                emit_diagnostic(ctx, literal.span);
+            }
+            AstKind::TemplateLiteral(literal)
+                if !is_tagged_template_expression(ctx, node, literal.span) =>
+            {
+                if literal
+                    .quasis
+                    .first()
+                    .unwrap()
+                    .value
+                    .raw
+                    .to_lowercase()
+                    .starts_with("javascript:")
```

Comment:
Looks like [ESLint](https://github.com/eslint/eslint/blob/9aa4df3f4d85960eee72923f3b9bfc88e62f04fb/lib/rules/utils/ast-utils.js#L230-L232) also check `quasis`'s length only be 1.

Reply:
done